src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -505,28 +504,10 @@ class SchedulerActions(
 object TaskStatusCollector {
   def collectTaskStatusFor(instances: Seq[Instance]): Seq[mesos.Protos.TaskStatus] = {
     instances.flatMap { instance =>
-      instance.tasksMap.values.withFilter(task => !task.isTerminal && !task.isReserved).map { task =>
-        task.status.mesosStatus.getOrElse(initialMesosStatusFor(task, instance.agentInfo))
+      instance.tasksMap.values.collect {
+        case task @ Task(_, _, Task.Status(_, _, Some(mesosStatus), _, _)) if !task.isTerminal =>
```

Comment:
Why did you choose to not filter out reserved anymore?

There's still two cases a task is in Reserved: (1) when it's just been created alongside the reservation, and (2) once we receive a terminal status update, which will transition the task back to `Condition.Reserved`. The former case is caught by leaving out tasks w/o status. The latter is not  contained in `isTerminal`. Since the task was reported terminal we don't want to reconcile for it – right?

Reply:
Even if this is probably temporary – can you please add a comment to the `def collectTaskStatusFor` which explains that the OverdueTasksActor is taking care of unconfirmed tasks?

Reply:
Yeah, the fact that we use Reserved for tasks that are killed and tasks that haven't been launched yet is very confusing. Second this question about why it was removed.

Reply:
well when we land phase 2, that won't be the case anymore. But you're right, on master I cannot remove that :(